.github/workflows/test-coverage.yml

```diff
@@ -0,0 +1,430 @@
+                    --platform       ${{ matrix.image }}
+                    --incremental    $INCREMENTAL
+                    --base_ref       ${{ github.base_ref }}
+                    --head_ref       refs/pull/${{ github.event.pull_request.number }}/head:${{ github.event.pull_request.head.ref }}
+                    --repository     ${{ github.server_url }}/${{ github.repository }}
+              "
+
+      - name: Workflow dispatch
+        if:   github.event_name == 'workflow_dispatch'
+        run: ".github/workflows/root-ci-config/build_root.py
+                    --buildtype      ${{ matrix.config }}
+                    --platform       ${{ matrix.image }}
+                    --incremental    ${{ inputs.incremental }}
+                    --base_ref       ${{ inputs.base_ref }}
+                    --head_ref       ${{ inputs.head_ref }}
+                    --repository     ${{ github.server_url }}/${{ github.repository }}
+              "
+
+      - name: Nightly build
+        if:   github.event_name == 'schedule'
+        run: ".github/workflows/root-ci-config/build_root.py
+                    --buildtype      ${{ matrix.config }}
+                    --platform       ${{ matrix.image }}
+                    --incremental    false
+                    --base_ref       ${{ github.ref_name }}
+                    --repository     ${{ github.server_url }}/${{ github.repository }}
+              "
+            
+
+
+      - name: Update artifacts after push to release branch
+        if:   github.event_name == 'push'
+        run: ".github/workflows/root-ci-config/build_root.py
+                    --buildtype      ${{ matrix.config }}
+                    --platform       ${{ matrix.image }}
+                    --incremental    false
+                    --base_ref       ${{ github.ref_name }}
+                    --repository     ${{ github.server_url }}/${{ github.repository }}
+              "
+    
+      - name: Upload test results
+        if:   ${{ !cancelled() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name:  Test Results ${{ matrix.image }} ${{ matrix.config }}
+          path: /github/home/ROOT-CI/build/TestResults.xml
+      
+      # - name: Upload coverage results
+      #   if:   ${{ matrix.image == 'fedora38-cov' && !cancelled() }}
+      #   uses: actions/upload-artifact@v3
+      #   with:
+      #     name: Test Coverage ${{ matrix.image }} ${{ matrix.config }}
+      #     path: /tmp/workspace/build/CoverageXML.xml
+      # - name: Setup tmate session
+      #   if: failure() && runner.debug == '1'
+      #   uses: mxschmitt/action-tmate@v3
+      #   with:
+      #     sudo: false
+      #   # When debugging increase to a suitable value!
+      #   timeout-minutes: ${{ github.event.pull_request && 40 || 20 }}
+
+      - name: Upload coverage to Codecov
+        if: ${{ matrix.image == 'fedora38-cov' && !cancelled() }}
+        uses: codecov/codecov-action@v3
```

Comment:
How is this authenticated? I suspect this needs a token? But that token won't be exposed to `pull_request` builds. Please have a look [here](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/), where they build with minimal credentials (`pull_request`), store the result as an artifact (we don't even need that!), and finally run the step that needs more permissions (in our case: to read secrets) to work just on the generated artifact. We can simply access that file in the `pull_request_target` part after the `pull_request` part building, testing, and generating the XML file has finished.

Let's discuss next week!